dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/update/postprocess/ProgramNotificationPostProcessor.java

```diff
@@ -68,4 +82,31 @@ public void process( final Event event, final WorkContext ctx )
             }
         }
     }
+
+    private ProgramStageInstance getProgramStageInstance( WorkContextWithEvent workContextWithEvent )
```

Comment:
This seems a bit of an overkill, especially from a readability point of view. Why not use the `getOrDefault` method of the Map interface:

```
final ProgramStageInstance programStageInstance = getProgramStageInstance( ctx, event );

...

private ProgramStageInstance getProgramStageInstance( WorkContext ctx, Event event )
    {
        return ctx.getProgramStageInstanceMap().getOrDefault( event.getUid(),
            new ProgramStageInstanceMapper( ctx ).map( event ) );
    }
```

Reply:
Also, note that the `ProgramStageInstanceMap` uses `event.getUid()` as key. The original `ProgramNotificationPostProcessor` had a bug, since it was using `event.getEvent()`